File: app/src/main/java/io/github/erikcaffrey/kata_dagger2_mariokart/di/PrincessPeachComponent.java
package io.github.erikcaffrey.kata_dagger2_mariokart.di;

import dagger.Component;
import io.github.erikcaffrey.kata_dagger2_mariokart.view.ui.PrincessPeachActivity;
import javax.inject.Singleton;

@Singleton @Component(modules = { PrincessPeachApiModule.class })
public interface PrincessPeachComponent {

  void inject(PrincessPeachActivity princessPeachActivity);
}

File: app/src/main/java/io/github/erikcaffrey/kata_dagger2_mariokart/di/PrincessPeachComponent.kt
package io.github.erikcaffrey.kata_dagger2_mariokart.di

import dagger.Component
import io.github.erikcaffrey.kata_dagger2_mariokart.view.ui.PrincessPeachActivity
import javax.inject.Singleton

@Singleton
@Component(modules = [PrincessPeachApiModule::class])
interface PrincessPeachComponent {

  fun inject(princessPeachActivity: PrincessPeachActivity)
}
